pkgs/top-level/aliases.nix

```diff
@@ -661,6 +661,8 @@ mapAliases ({
   youtubeDL = youtube-dl;  # added 2014-10-26
   zdfmediathk = mediathekview; # added 2019-01-19
   gnome_user_docs = gnome-user-docs; # added 2019-11-20
+  # spidermonkey is not ABI upwards-ompatible, so only allow this for nix-shell
+  spidermonkey = spidermonkey_78; # added 2020-10-09
```

Comment:
A brief note that aliases.nix are enabled by default for users of nixpkgs, and their use is only prohibited from within nixpkgs itself. In other words, `nix-shell` isn't really the determining factor here.